clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java

```diff
@@ -1310,16 +1313,23 @@ private CompletedFetch initializeCompletedFetch(CompletedFetch nextCompletedFetc
         return completedFetch;
     }
 
-    private void handleOffsetOutOfRange(long fetchOffset,
+    private void handleOffsetOutOfRange(FetchPosition fetchPosition,
                                         TopicPartition topicPartition,
                                         String reason) {
         if (subscriptions.hasDefaultOffsetResetPolicy()) {
-            log.info("Fetch offset {} is out of range for partition {}, resetting offset",
-                topicPartition, fetchOffset);
+            log.info("Fetch offset epoch {} is out of range for partition {}, resetting offset",
+                fetchPosition, topicPartition);
             subscriptions.requestOffsetReset(topicPartition);
         } else {
-            throw new OffsetOutOfRangeException(Collections.singletonMap(
-                topicPartition, fetchOffset), reason);
+            Map<TopicPartition, Long> offsetOutOfRangePartitions =
+                Collections.singletonMap(topicPartition, fetchPosition.offset);
+            String errorMessage = String.format("Offsets out of range " +
+                "with no configured reset policy for partitions: %s" +
+                ", for fetch offset: %d, " +
+                "root cause: %s",
+                offsetOutOfRangePartitions, fetchPosition.offset, reason);
+            log.error(errorMessage);
```

Comment:
Could we make it `info` in both cases? If the application has not set a reset policy, then they are expecting to handle the exception.